core/sql/optimizer/ScanOptimizer.cpp

```diff
@@ -374,8 +399,21 @@ static NABoolean checkMDAMadditionalRestriction(const ColumnOrderList& keyPredsB
 
      }
    }     
-   if(presentKeyColumns > noOfmissingKeyColumns) return TRUE;
-   return FALSE;
+   switch ( strategy ) {
+     case MAJORITY_WITH_PREDICATES:
+        return (presentKeyColumns > noOfmissingKeyColumns);
```

Comment:
I agree and we are going to use the new strategy (total UEC) as the default
one.

Thanks

On Mon, Nov 30, 2015 at 11:23 AM, DaveBirdsall <notifications@github.com>
wrote:

> In core/sql/optimizer/ScanOptimizer.cpp
> <https://github.com/apache/incubator-trafodion/pull/195#discussion_r46173600>
> :
>
> > @@ -374,8 +399,21 @@ static NABoolean checkMDAMadditionalRestriction(const ColumnOrderList& keyPredsB
> >
> >       }
> >     }
> > -   if(presentKeyColumns > noOfmissingKeyColumns) return TRUE;
> > -   return FALSE;
> > +   switch ( strategy ) {
> > +     case MAJORITY_WITH_PREDICATES:
> > +        return (presentKeyColumns > noOfmissingKeyColumns);
>
> I am not sure I see the point of this strategy. Suppose I have a 6 column
> key and a query with predicates on 4 columns. It makes a huge difference
> whether the predicates are on the leading columns (better) vs. the trailing
> columns (worse). This heuristic does not take the column order into account.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/195/files#r46173600>.
>



-- 
Regards, --Qifan